### PR TITLE
Feat: portal cubes

### LINKED ISSUE #230: Implement House Portals
## Description

This issue is to track the implementation of house portals in the LoadUgcMapHandler.

## Tasks

- [ ] Handle adding portal cube and spawn point on `RequestCubeHandler.cs`
- [ ] Implement a method to check if a plot has portal entry points and use it in `LoadUgcMapHandler.cs`

## Related Pull Request

This issue was created in response to a comment in PR #229.

### PR DESCRIPTION
**Pull request by AngeloTadeucci**
- Resolves #230

## Summary by CodeRabbit

- **New Features**
	- Enhanced `HomeLayoutCube` and `UgcMapCube` classes with new properties: `HousingCategory` and `CubeSettings`.
	- Introduced `FunctionCubeMetadata` and related storage and mapping classes to manage function cube data.
	- Added new methods for managing portals in the `FieldManager` and improved cube placement logic in the `HousingManager`.
	- Implemented functionality for migrating game sessions to specified home locations.

- **Bug Fixes**
	- Improved checks for cube placement restrictions based on housing categories.

- **Documentation**
	- Updated migration scripts to reflect new database structures for home layouts and portal settings.

### CODE REVIEW

**Comment by AngeloTadeucci** [comment]
@coderabbitai configuration

File: Maple2.Database/Model/Map/UgcMapCube.cs [hunk 1]
```diff
@@ -16,13 +17,19 @@ internal class UgcMapCube {
     public float Rotation { get; set; }
 
     public int ItemId { get; set; }
+    public HousingCategory HousingCategory { get; set; }
+    public CubePortalSettings? PortalSettings { get; set; }
```
Comment (Zintixx) [inline Maple2.Database/Model/Map/UgcMapCube.cs:21]: I am wondering if this should be expanded for various cube that has its own settings, such as megaphones or nuturing npcs. Expanded in the same sense as ItemAppearance in Item

File: Maple2.Model/Game/Cube/PlotCube.cs [hunk 2]
```diff
@@ -1,19 +1,60 @@
...
+    /// Reverse and transform to hexadecimal as string: '1FFFF';
+    /// Convert the string above to long: 65535.
+    /// </summary>
+    private long AsHexadecimal() {
+        string coordRevertedAsString = $"{Position.Z:X2}{Position.Y:X2}{Position.X:X2}";
+        return Convert.ToInt64(coordRevertedAsString, 16);
+    }
+}
+
+public enum InteractCubeState {
```
Comment (Zintixx) [inline Maple2.Model/Game/Cube/PlotCube.cs:56]: Move this Enum to Model/Enum

**Review by Zintixx — CHANGES_REQUESTED** [review]

File: Maple2.Database/Model/Map/CubeHelper.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using Maple2.Model.Enum;
+using Maple2.Model.Game;
+
+namespace Maple2.Database.Model;
+
+internal static class CubeHelper {
+    public static CubeSettings? GetCubeSettings(PlotCube cube) {
```
Comment (Zintixx) [inline Maple2.Database/Model/Map/CubeHelper.cs:7]: Do we really need this? Or can we just do the conversion within the model? Similar to how ItemAppearance is done.

**Review by Zintixx — APPROVED** [review]

**Pull request merged**